### ISSUE TITLE
Setting `OCAMLRUNPARAM=` doesn't seem to turn off exception backtraces

### ISSUE THREAD

**Issue by abliss** [realworldocaml/book#3377]
See the claim here: https://dev.realworldocaml.org/error-handling.html#:~:text=you%20can%20turn%20backtraces%20off

But as the text in the output blurb shows, it doesn't work. The text claims "The resulting error message is considerably less informative" but actually it's identical to the previous blurb (without setting `OCAMLRUNPARAM=`). 

I verified this on my local install. The backtrace is still there.

**Comment by abliss** [comment]
Also, the benchmark results that follow don't seem to line up with the commentary on them. The variation within the first set is not what the text claims, and the second set is much slower when the text claims it should be faster.

**Comment by yminsky** [comment]
Yeah, I can confirm both issues.  This kind of example is more likely to rot, because it's not run every time, since the results are non-deterministic.  And indeed, it has rotted:

```
[examples]$ dune runtest --profile non-deterministic
Entering directory '/home/yminsky/Documents/code/rwo'
        dune alias book/error-handling/examples/blow_up (exit 1)
(cd _build/default/book/error-handling/examples && ../../../../install/default/bin/dune build @all @runtest --root ./correct/blow_up)
Entering directory 'correct/blow_up'
File "/home/yminsky/Documents/code/rwo/_build/install/default/lib/stdio/dune-package", line 1, characters 11-14:
1 | (lang dune 2.7)
               ^^^
Error: Version 2.7 of the dune language is not supported.
Supported versions of this extension in version 2.7 of the dune language:
- 1.0 to 1.12
- 2.0 to 2.6
        dune alias book/error-handling/examples/exn_cost (exit 1)
(cd _build/default/book/error-handling/examples && ../../../../install/default/bin/dune build @all @runtest --root ./correct/exn_cost)
Entering directory 'correct/exn_cost'
File "/home/yminsky/Documents/code/rwo/_build/install/default/lib/core_bench/dune-package", line 1, characters 11-14:
1 | (lang dune 2.7)
               ^^^
Error: Version 2.7 of the dune language is not supported.
Supported versions of this extension in version 2.7 of the dune language:
- 1.0 to 1.12
- 2.0 to 2.6
```

I'm very confused where this error is coming from, since dune 2.7 isn't mentioned in any files. @avsm @NathanReb can you take a look at this one?

Also, it seems like the behavior of Base has changed such that the text is wrong. I'll dig into this separately.

**Comment by yminsky** [comment]
Oh, and the second change in behavior is clear enough. Once you open Base, the default is effectively turned to backtraces on.  It's not enough to set OCAMLRUNPARAM to the empty string to change it (that's actually super weird, if you stop to think about it.)  Instead, you have to write `OCAMLRUNPARAM=b=0` to disable backtraces, which makes sense.  I'm fixing it, and once I can rerun the benchmarks, I'll push the fix.

Thanks for the catch Adam!

**Comment by yminsky** [comment]
The draft is done, and ignore my intermediate complaint about dune 2.7. I had neglected to do `eval $(opam env)` in the right spot.

**Comment by yminsky** [comment]
Fixed by #3381
